core/string/string_name.cpp

```diff
@@ -33,38 +33,31 @@
 #include "core/os/os.h"
 #include "core/string/print_string.h"
 
-bool StringName::_Data::operator==(const String &p_name) const {
-	return name == p_name;
-}
-
-bool StringName::_Data::operator!=(const String &p_name) const {
-	return !operator==(p_name);
-}
+struct StringName::Table {
+	constexpr static uint32_t TABLE_BITS = 16;
+	constexpr static uint32_t TABLE_LEN = 1 << TABLE_BITS;
+	constexpr static uint32_t TABLE_MASK = TABLE_LEN - 1;
 
-bool StringName::_Data::operator==(const char *p_name) const {
-	return name == p_name;
-}
-
-bool StringName::_Data::operator!=(const char *p_name) const {
-	return !operator==(p_name);
-}
+	static inline _Data *table[TABLE_LEN];
```

Comment:
Does `static inline _Data *table[TABLE_LEN] = {};` not 0 initialize it, so we could remove the setup call altogether ?

Reply:
It would. But it's out of scope. It probably makes sense to allocate at runtime anyway to avoid the 500kb bloating the binary.

Reply:
I thought standard forced static arrays to be zero initialized anyway, but I'm not sure with inline. And it would probably be put in the bss section, so it would not bloat anything (but making sure no compiler options is messing with that would be the _no fun_ part)

Reply:
I'm pretty sure bloaty told me the table made up 500kb of the final binary. I was planning on testing this again (along with potential speed impact) when I make the according PR.